pandas_gbq/schema.py

```diff
@@ -31,3 +31,32 @@ def generate_bq_schema(dataframe, default_type="STRING"):
         )
 
     return {"fields": fields}
+
+
+def update_schema(schema_old, schema_new):
+    """
+    Given an old BigQuery schema, update it with a new one.
+
+    Where a field name is the same, the new will replace the old. Any
+    new fields not present in the old schema will be added.
+
+    Arguments:
+        schema_old: the old schema to update
+        schema_new: the new schema which will overwrite/extend the old
+    """
+    old_fields = schema_old["fields"]
+    new_fields = schema_new["fields"]
+    output_fields = old_fields.copy()
+
+    field_indices = {field["name"]: i for i, field in enumerate(output_fields)}
```

Comment:
Do we need `enumerate` here? Are we using a dict as a list?

I don't think it's that important though - the tests look good and so I'm comfortable with the code.

Reply:
It's just to get the indices of existing fields by name. Without getting this up front, for every updated field, we have to go through the whole list and look for which one to update. But since `enumerate` gives us the indices, we can just replace the relevant list element directly.

Note that `field_indices` is only being used for updating fields, it's never returned or anything.

Reply:
👍